### PR TITLE
[otel-integration] add upgrading docs

### PR DESCRIPTION
**Pull request by povilasv**
# Description

Fixes #ES-147

# How Has This Been Tested?

# Checklist:
- [ ] I have updated the relevant Helm chart(s) version(s)
- [ ] I have updated the relevant component changelog(s)
- [x] This change does not affect any particular component (e.g. it's CI or docs change)

### CODE REVIEW

**Review by oded-dd — COMMENTED** [review]
I think we should emphasize the fact that this is a `breaking change`

File: otel-integration/UPGRADING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+## 0.0.32 to 0.0.33
+
+Opentelemetry Integration v0.33 has removed Kube State Metrics deployment, as the needed resource attributes and metrics are now part of OpenTelemetry Kubernetes Cluster Receiver.
+
+We removed `metricstransform/kube-extra-metrics` processor, as it was transforming `kube-state-metrics` metrics. And added two processors `metricstransform/k8s-dashboard` and `transform/k8s-dashboard` which transforms OTEL format into the format which Coralogix Kubernetes Dashboard uses.
+
+If you manually configured the opentelemetry-cluster-collectors metrics pipeline, you will need add `metricstransform/k8s-dashboard` and `transform/k8s-dashboard` processors to your pipeline.
+
+Previously, we had cluster collector's metrics pipeline looked like this:
```
Comment (matej-g) [inline otel-integration/UPGRADING.md:15]:
```suggestion
Previously, we had cluster collector's metrics pipeline that looked like this:
```

File: otel-integration/UPGRADING.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+If the version you want to upgrade to is not listed here, then there is nothing to do for you.
+Just upgrade and enjoy.
+
+## 0.0.32 to 0.0.33
+
+Opentelemetry Integration v0.33 has removed Kube State Metrics deployment, as the needed resource attributes and metrics are now part of OpenTelemetry Kubernetes Cluster Receiver.
+
+We removed `metricstransform/kube-extra-metrics` processor, as it was transforming `kube-state-metrics` metrics. And added two processors `metricstransform/k8s-dashboard` and `transform/k8s-dashboard` which transforms OTEL format into the format which Coralogix Kubernetes Dashboard uses.
+
+If you manually configured the opentelemetry-cluster-collectors metrics pipeline, you will need add `metricstransform/k8s-dashboard` and `transform/k8s-dashboard` processors to your pipeline.
```
Comment (matej-g) [inline otel-integration/UPGRADING.md:13]:
```suggestion
If you manually configured the opentelemetry-cluster-collectors metrics pipeline, you will need to add `metricstransform/k8s-dashboard` and `transform/k8s-dashboard` processors to your pipeline.
```

**Review by matej-g — APPROVED** [review]
Small nits, looks good 👍

**Pull request merged**